src/App.tsx

```diff
@@ -19,55 +47,56 @@ export const App: React.FC = () => {
               placeholder="Enter a part of the name"
               className="input"
               data-cy="search-input"
+              value={query}
+              onChange={handleQueryChange}
+              onFocus={() => setFocusedQuery(true)}
+              onBlur={() => setTimeout(() => setFocusedQuery(false), 300)}
             />
           </div>
 
-          <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
-            <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
+          {focusedQuery && (
+            <div
+              className="dropdown-menu"
+              role="menu"
+              data-cy="suggestions-list"
+            >
+              <div className="dropdown-content">
+                {filteredNames.map((person) => {
+                  const thatPerson = highlatedPerson?.slug === person.slug;
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Petronella de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Hercke</p>
+                  return (
+                    <div
+                      key={person.slug}
+                      onClick={() => handleClick(person)}
+                      className={`dropdown-item is-clickable ${thatPerson && 'is-active'}`}
+                      data-cy="suggestion-item"
+                    >
+                      <p className={`has-text-${thatPerson ? 'danger' : 'link'}`}>
```

Comment:
Use [classnames](https://www.npmjs.com/package/classnames) library to add classes conditionally
Fix all cases